### PR TITLE
atlasexec: change migrate push context type

### PR DESCRIPTION
**Pull request by dorav**

### CODE REVIEW

**Review by a8m — APPROVED** [review]

File: atlasexec/atlas_test.go [hunk 1]
```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
...
+				} `json:"input"`
+			}
+		}
+		httpTest struct {
+			payloads []graphQLQuery
+			srv      *httptest.Server
+		}
+	)
+	token := "123456789"
+	newHttpTest := func() (*httpTest, string) {
```
Comment (a8m) [inline atlasexec/atlas_test.go:451]:
```suggestion
	newServer := func() (*httpTest, string) {
```

Comment (a8m) [reply atlasexec/atlas_test.go:451]: If you prefer the same naming, use `http`/`HTTP` rather than `Http`.

File: atlasexec/atlas_test.go [hunk 2]
```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
...
+				require.Equal(t, "Bearer "+token, r.Header.Get("Authorization"))
+				var query graphQLQuery
+				require.NoError(t, json.NewDecoder(r.Body).Decode(&query))
+				tt.payloads = append(tt.payloads, query)
+				if strings.Contains(query.Query, "pushDir") {
+					fmt.Fprint(w, `{"data":{"pushDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
+				if strings.Contains(query.Query, "syncDir") {
+					fmt.Fprint(w, `{"data":{"syncDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
```
Comment (a8m) [inline atlasexec/atlas_test.go:464]: If you create a server per test, you can mock the body in the creation. e.g., 
```
newServer(`{"data": {"syncDir": "value"}}`)
```

File: atlasexec/atlas_test.go [hunk 3]
```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
...
+			require.NoError(t, err)
+			require.NotEqual(t, "1234", got)
+			require.Len(t, tt.payloads, 2)
+			p := &tt.payloads[1]
+			require.Contains(t, p.Query, "syncDir")
+			err = json.Unmarshal(p.Variables, &p.SyncDir)
+			require.NoError(t, err)
+			require.Equal(t, "test-dir-slug", p.SyncDir.Input.Slug)
+			require.Equal(t, "SQLITE", p.SyncDir.Input.Driver)
+			require.NotEmpty(t, p.SyncDir.Input.Dir)
```
Comment (a8m) [inline atlasexec/atlas_test.go:516]: My suggestion - no need to copy-paste everything. Test only the `context` change. 
```suggestion
```

File: atlasexec/atlas_test.go (hunk 2 above)
Comment (dorav) [reply atlasexec/atlas_test.go:464]: I think it's best to keep it this way, if more tests are added in the future for the same use-case (more tests that need 'syncDir' for example, the response is already built-in and working)

File: atlasexec/atlas_test.go [hunk 4]
```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
...
+			DirURL: "file://testdata/migrations",
+			Name:   "test-dir-slug",
+			Env:    "test",
+		}
+		t.Run("with context", func(t *testing.T) {
+			tt, atlasConfigURL := newHTTPTest()
+			params.ConfigURL = atlasConfigURL
+			got, err := c.MigratePush(context.Background(), params)
+			require.NoError(t, err)
+			require.NotEqual(t, "1234", got)
```
Comment (dorav) [inline atlasexec/atlas_test.go:492]: rm, placeholder

File: atlasexec/atlas_test.go [hunk 5]
```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
...
+			DevURL:    "sqlite://file?mode=memory",
+			DirURL:    "file://testdata/migrations",
+			Name:      "test-dir-slug",
+			Context:   inputContext,
+			Env:       "test",
+			Tag:       "this-is-my-tag",
+		}
+		got, err := c.MigratePush(context.Background(), params)
+		require.NoError(t, err)
+		require.NotEqual(t, "1234", got)
```
Comment (dorav) [inline atlasexec/atlas_test.go:534]: rm, placeholder

**Pull request merged**